src/rrt/Tree.hpp

```diff
@@ -309,7 +350,21 @@ class Tree {
         }
 
         // Add a node to the tree for this state
+        int lengthT = sizeof(intermediateState) / sizeof(0.0);
+        double* data = new double[lengthT];
+        if (NULL == _TToArray) {
+            for (int i = 0; i < lengthT; i++) {
+                data[i] = intermediateState[i];
+            }
+        } else {
+            data = _TToArray(intermediateState);
```

Comment:
You mean combine TToArray and ArrayToT into one function that would look something like convert(T state, double* data)?